source/slang/slang-check-expr.cpp

```diff
@@ -1355,6 +1355,162 @@ namespace Slang
         }
     }
 
+    RefPtr<Expr> SemanticsVisitor::CheckMatrixSwizzleExpr(
+        MemberExpr* memberRefExpr,
+        RefPtr<Type>      baseElementType,
+        IntegerLiteralValue baseElementRowCount,
+        IntegerLiteralValue baseElementColCount)
+    {
+        RefPtr<MatrixSwizzleExpr> swizExpr = new MatrixSwizzleExpr();
+        swizExpr->loc = memberRefExpr->loc;
+        swizExpr->base = memberRefExpr->baseExpression;
+
+        int elementIndices[4][2];
+        int elementCount = 0;
+
+        bool anyDuplicates = false;
+        int oneIndexed = -1;
```

Comment:
Nit pick: This variable name reads like a boolean, and it sort of looks like it works that way until later it's used to subtract to make 1 indexed value zero indexed. I might be better to be named as zeroIndexOffset, zeroIndexFix or something like that.